megamek/src/megamek/common/BattleArmor.java

```diff
@@ -702,9 +702,11 @@ public HitData getTrooperAtLocation(HitData hit, Entity transport) {
                 }
                 break;
             case Tank.LOC_REAR:
-                if ((getInternal(5) > 0) && (getInternal(6) > 0)) {
+                //These troopers only exist when you have Clan and CS/WoB units, so we need
+                //to ensure the array is large enough before checking whether they're dead.
+                if (((locations() >= 6) && (getInternal(5) > 0)) && ((locations() >= 7) && (getInternal(6) > 0))) {
                     loc = Compute.randomInt(2) + 5;
-                } else if (getInternal(5) > 0) {
+                } else if ((locations() >= 6) && (getInternal(5) > 0)) {
                     loc = 5;
```

Comment:
Seems like you'd want loc to be 6 here and 5 in the else block?

Reply:
The arrays have 1 extra location for squad

Reply:
I think I'm having trouble understanding the logic then. 

If we've got 7 troopers then we either pick trooper 6 or 7
If we've got 6 troopers then we pick trooper 6
Otherwise we pick trooper 7?

Reply:
I made a couple of adjustments to the test logic that I think will take care of the issues here.